packages/relay-compiler/generic/core/RelayWatchmanClient.js

```diff
@@ -17,6 +17,15 @@ const watchman = require('fb-watchman');
 class RelayWatchmanClient {
   _client: any;
 
+  static createIfAvailable(): Promise<?RelayWatchmanClient> {
+    return new Promise((resolve) => {
+      const client = new RelayWatchmanClient();
+      client.on('error', () => resolve(null));
```

Comment:
I'm curious if there are certain kinds of errors which should cause this to be determined as "unavailable" as opposed to considering all errors? I'm interested in what you saw during investigating writing this PR that led you to this